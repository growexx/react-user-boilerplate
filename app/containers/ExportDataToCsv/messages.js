/*
 * ExportDataToCsv Messages
 *
 * This contains all the text for the ExportDataToCsv container.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.containers.ExportDataToCsv';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the ExportDataToCsv container!',
  },
});
