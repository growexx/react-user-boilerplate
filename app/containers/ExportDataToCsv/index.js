/**
 *
 * ExportDataToCsv
 *
 */

import React from 'react';
import { Helmet } from 'react-helmet';
import { Button, Table } from 'antd';
import { FormattedMessage } from 'react-intl';
import { exportJsonAsCSV } from '../../utils/csvExport';
import { StyledButton, StyledExport } from './StyledExport';
import {
  CSV_FILE_NAME,
  FIELDS_FOR_CSV,
  TABLE_COLUMNS,
  TABLE_DATA,
} from './constants';
import messages from './messages';
class ExportDataToCsv extends React.Component {
  state = {
    selectedRows: [],
  };

  rowSelection = {
    onChange: (selectedRowKeys, selectedRows) => {
      // eslint-disable-next-line no-console
      console.log(
        `selectedRowKeys: ${selectedRowKeys}`,
        'selectedRows: ',
        selectedRows,
      );
      this.setState({
        selectedRows,
      });
    },
  };

  exportData = () => {
    exportJsonAsCSV(
      this.state.selectedRows || [],
      FIELDS_FOR_CSV,
      CSV_FILE_NAME,
    );
  };

  render() {
    return (
      <div>
        <Helmet>
          <title>ExportDataToCsv</title>
          <meta name="description" content="Description of ExportDataToCsv" />
        </Helmet>
        <StyledExport>
          <StyledButton>
            <Button
              type="primary"
              onClick={this.exportData}
              disabled={this.state.selectedRows.length === 0}
            >
              <FormattedMessage {...messages.exportData} />
            </Button>
          </StyledButton>
          <Table
            pagination={false}
            rowSelection={{
              type: 'checkbox',
              ...this.rowSelection,
            }}
            columns={TABLE_COLUMNS}
            dataSource={TABLE_DATA}
          />
        </StyledExport>
      </div>
    );
  }
}

export default ExportDataToCsv;
