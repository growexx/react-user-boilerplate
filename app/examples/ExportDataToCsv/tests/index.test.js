/**
 *
 * Tests for ExportDataToCsv
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import history from 'utils/history';
import { browserHistory } from 'react-router-dom';
import ExportDataToCsv from '../index';
import configureStore from '../../../configureStore';
let store;
const componentWrapper = () =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <ExportDataToCsv />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );

describe('<ExportDataToCsv />', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
  it('should trigger export', () => {
    const { container } = componentWrapper();
    const checkbox = container.querySelector('input');
    const element = checkbox;
    fireEvent.click(element);
    const { getByTestId } = componentWrapper();
    const buttonElement = getByTestId('ExportButton');
    fireEvent.click(buttonElement);
    expect(buttonElement.tagName).toEqual('BUTTON');
  });
});
