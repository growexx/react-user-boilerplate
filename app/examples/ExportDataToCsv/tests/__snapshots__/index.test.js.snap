// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExportDataToCsv /> Should render and match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 button {
  margin-bottom: 12px;
}

.c1 {
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        class="ant-btn ant-btn-primary"
        data-testid="ExportButton"
        disabled=""
        type="button"
      >
        <span>
          Export Data
        </span>
      </button>
    </div>
    <div
      class="ant-table-wrapper"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table ant-table-scroll-horizontal"
            data-testid="DataTable"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
                style="overflow-x: auto; overflow-y: hidden;"
              >
                <table
                  style="width: 350px; min-width: 100%; table-layout: auto;"
                >
                  <colgroup>
                    <col
                      class="ant-table-selection-col"
                    />
                  </colgroup>
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell ant-table-selection-column"
                      >
                        <div
                          class="ant-table-selection"
                        >
                          <label
                            class="ant-checkbox-wrapper"
                          >
                            <span
                              class="ant-checkbox"
                            >
                              <input
                                class="ant-checkbox-input"
                                type="checkbox"
                                value=""
                              />
                              <span
                                class="ant-checkbox-inner"
                              />
                            </span>
                          </label>
                        </div>
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        Name
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        Age
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        Address
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      aria-hidden="true"
                      class="ant-table-measure-row"
                      style="height: 0px; font-size: 0;"
                    >
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="1"
                    >
                      <td
                        class="ant-table-cell ant-table-selection-column"
                      >
                        <label
                          class="ant-checkbox-wrapper"
                        >
                          <span
                            class="ant-checkbox"
                          >
                            <input
                              class="ant-checkbox-input"
                              type="checkbox"
                              value=""
                            />
                            <span
                              class="ant-checkbox-inner"
                            />
                          </span>
                        </label>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        John Brown
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        32
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        New York No. 1 Lake Park
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="2"
                    >
                      <td
                        class="ant-table-cell ant-table-selection-column"
                      >
                        <label
                          class="ant-checkbox-wrapper"
                        >
                          <span
                            class="ant-checkbox"
                          >
                            <input
                              class="ant-checkbox-input"
                              type="checkbox"
                              value=""
                            />
                            <span
                              class="ant-checkbox-inner"
                            />
                          </span>
                        </label>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        Jim Green
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        42
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        London No. 1 Lake Park
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="3"
                    >
                      <td
                        class="ant-table-cell ant-table-selection-column"
                      >
                        <label
                          class="ant-checkbox-wrapper"
                        >
                          <span
                            class="ant-checkbox"
                          >
                            <input
                              class="ant-checkbox-input"
                              type="checkbox"
                              value=""
                            />
                            <span
                              class="ant-checkbox-inner"
                            />
                          </span>
                        </label>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        Joe Black
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        32
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        Sidney No. 1 Lake Park
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
